lxd/images.go

```diff
@@ -1617,6 +1617,23 @@ func imagesGet(d *Daemon, r *http.Request) response.Response {
 	filterStr := r.FormValue("filter")
 
 	s := d.State()
+	err := s.DB.Cluster.Transaction(r.Context(), func(ctx context.Context, tx *db.ClusterTx) error {
+		hasImages, err := dbCluster.ProjectHasImages(ctx, tx.Tx(), projectName)
+		if err != nil {
+			return err
+		}
+
+		if !hasImages {
+			projectName = api.ProjectDefaultName
```

Comment:
Ah yes I missed that. This won't change anything as `doImagesGet` repeats the call to get the effective project. The only difference here is that the effective project is already being passed in. I'll change it back to the query parameter, this PR is just for auth and shouldn't be changing this.

Ideally we will refactor the image db functions so that the effective project can be passed in because they are a little convoluted at the moment.

Reply:
Fixed.